$MirOS: ports/security/gnupg/patches/patch-configure_ac,v 1.4 2008/04/01 15:47:02 tg Exp $
--- configure.ac.orig	Wed Mar 26 17:26:26 2008
+++ configure.ac	Tue Aug 26 19:20:43 2008
@@ -28,9 +28,6 @@ min_automake_version="1.9.3"
 m4_define([my_version], [1.4.9])
 m4_define([my_issvn], [no])
 
-m4_define([svn_revision], m4_esyscmd([echo $((svn info 2>/dev/null \
-          || echo 'Revision: 0')|sed -n '/^Revision:/ s/[^0-9]//gp'|head -1)| \
-          tr -d '\n']))
 AC_INIT([gnupg], my_version[]m4_if(my_issvn,[yes],[-svn[]svn_revision]),
         [bug-gnupg@gnu.org])
 # Set development_version to yes if the minor number is odd or you
@@ -826,12 +823,12 @@ AC_DEFINE_UNQUOTED(PRINTABLE_OS_NAME, "$
 #
 NAME_OF_DEV_RANDOM="/dev/random"
 NAME_OF_DEV_URANDOM="/dev/urandom"
-case "${host}" in
-    *-openbsd*)
+if test -e /dev/srandom; then
         NAME_OF_DEV_RANDOM="/dev/srandom"
-        NAME_OF_DEV_URANDOM="/dev/urandom"
-        ;;
-esac
+fi
+if test -e /dev/arandom; then
+	NAME_OF_DEV_URANDOM="/dev/arandom"
+fi
 AC_DEFINE_UNQUOTED(NAME_OF_DEV_RANDOM, "$NAME_OF_DEV_RANDOM",
                    [defined to the name of the strong random device])
 AC_DEFINE_UNQUOTED(NAME_OF_DEV_URANDOM, "$NAME_OF_DEV_URANDOM",
@@ -979,6 +976,16 @@ if test x"$use_sha512" = xyes ; then
 fi
 
 AM_CONDITIONAL(USE_SHA512, test x"$use_sha512" = xyes)
+
+AC_CHECK_HEADERS([sys/sysctl.h], [], [], [#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif])
+AC_CHECK_HEADERS([uvm/uvm_swap_encrypt.h], [], [], [#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#ifdef HAVE_SYS_SYSCTL_H
+#include <sys/sysctl.h>
+#endif])
 
 dnl Checks for library functions.
 AC_CHECK_DECLS(getpagesize)
